internal/broker/broker.go

```diff
@@ -204,24 +211,19 @@ func (b *Broker) GetAuthenticationModes(sessionID string, supportedUILayouts []m
 	_, err = os.Stat(session.cachePath)
 	tokenExists := err == nil
 
-	// Checks if the provider is accessible and if device authentication is supported
-	providerReachable := true
-	r, err := http.Get(strings.TrimSuffix(b.auth.providerURL, "/") + "/.well-known/openid-configuration")
-	// This means the provider is not available or something bad happened, so we assume no connection.
-	if err != nil || r.StatusCode != http.StatusOK {
-		providerReachable = false
+	endpoints := make(map[string]struct{})
+	if session.authCfg.provider != nil {
+		if session.authCfg.provider.Endpoint().DeviceAuthURL != "" {
+			endpoints["device_auth"] = struct{}{}
```

Comment:
I think in a longer term, we should actually use a struct with common capabilities to declare them.